### ISSUE TITLE
Keyboard Layout widget does not show the current layout on latest Manjaro

### ISSUE THREAD

**Issue by karevn** [qtile/qtile#1540]
I have Manjaro with qtile, and tried to set up KeyboardLayout widget. Unfortunately, it does not react to keyboard layout change when using a hot key to change the layout. 
So, steps to reproduce:
1. Have latest manjaro, `us,ru` keyboards.
2. Press a layout switch shortcut (win + space in my case). XKB switches to the Russian layout at this moment.
3. I expect the Keyboard Layout widget to display the RU layout, but it does not.

And one more problem:
1. Same setup
2. I click the widget text (which is US)
3. The keyboard switches to the Russian layout, and so does the Keyboard Layout widget. 
4. But the system hotkey does not work anymore.

A quick peek into the code has revealed that qtile tries to execute 
```bash
setxkbmap -verbose 10
```
which returns the next:
```
Setting verbose level to 10
locale is C
Trying to load rules file ./rules/evdev...
Trying to load rules file /usr/share/X11/xkb/rules/evdev...
Success.
Applied rules from evdev:
rules:      evdev
model:      pc105
layout:     us,ru
variant:    ,
options:    grp:win_space_toggle
Trying to build keymap using the following components:
keycodes:   evdev+aliases(qwerty)
types:      complete
compat:     complete
symbols:    pc+us+ru:2+inet(evdev)+group(win_space_toggle)
geometry:   pc(pc105)
```

Then, it looks like regexps used to parse the layout do not work correctly

**Comment by zordsdavini** [comment]
the problem is that widget has it's own currentLayout and doesn't change it if it is changed from system side. I tried to add options for switching but it looks it works not correct and each switch do new `setxkbmap`. I solved that by custom keybinding to swith widget layout and not through setxkbmap. You can check here https://github.com/zordsdavini/dotfiles/blob/master/.config/qtile/config.py

**Comment by daroczig** [comment]
@zordsdavini this way can you set a custom key in `options`, eg `grp:rctrl_switch` to have a shortcut to temp change the layout? It doesn't seem to work for me. Then if I run something like `setxkbmap -layout us,hu -option grp:rctrl_switch` on the command line -- that makes it possible, but then the widget will not update. I might need to drop `widget.KeyboardLayout` and just use a `widget.GenPollText` to show the current layout.

**Comment by zordsdavini** [comment]
@daroczig yes, it has bug and I don't find time to look at that and fix. I think the problem is that we quering current layout with `setxkbmap -verbose 10` while we should use `setxkbmap -query`. Next `-options` should be set once. If you have time - you can look at that :) And yes - you can try with `widget.GenPollText` as alternative

**Comment by zordsdavini** [comment]
reviewed `setxkbmap`. Dumb tool but does its work: `set`. There is no way to return current layout from it. So, that job is left to widget. But during set it sets only one layout, so switching is impossible because set layout is only one.  

And back to @karevn issue. From output I see that you have used setxkbmap from console. This sadly brakes all flow and widget can't work any more correctly (in my case even Qtile reload didn't help).
So, to continue, we need config to see how you set or something more

**Comment by ngc-bg** [comment]
Sorry for bringing up this already closed issue, still, I am experiencing it...

I am setting my setxkbmap trough autostart.sh script:
`setxkbmap -layout us,bg -variant ,phonetic -option grp:alt_shift_toggle`

which is working fine for instance, when using polybar with their internal/xkeyboard module. With Qtile I am not willing to substitute the internal bar with polybar, so I have to find a workaround.

Printing verbose setxkbmap:
```
Setting verbose level to 10
locale is C
Trying to load rules file ./rules/evdev...
Trying to load rules file /usr/share/X11/xkb/rules/evdev...
Success.
Applied rules from evdev:
rules:      evdev
model:      pc105
layout:     us,bg
variant:    ,phonetic
options:    grp:alt_shift_toggle,terminate:ctrl_alt_bksp
Trying to build keymap using the following components:
keycodes:   evdev+aliases(qwerty)
types:      complete
compat:     complete
symbols:    pc+us+bg(phonetic):2+inet(evdev)+group(alt_shift_toggle)+terminate(ctrl_alt_bksp)
geometry:   pc(pc105)
```
I'll try to configure what @daroczig suggested - to use widget.GenPollText to show the current layout.
_And btw, I didn't found a way to convince config.py to accept alt+shift to be toggle/terminate keycombo..._

**Comment by renatojobal** [comment]
For me it works fine with this config: 
```
        widget.KeyboardLayout(
            font="Noto Sans Bold",
            foreground=colors[5],
            background=colors[10],
            fontsize=12,
            configured_keyboards=["us", "es"],
            mouse_callback={'Button1': lambda : lazy.widget["keyboardlayout"].next_keyboard()},
        ),
```
